openedx/adg/lms/applications/tests/test_helpers.py

```diff
@@ -550,16 +552,73 @@ def test_get_course_card_information_of_a_course_with_prerequisite(
     if is_course_locked:
         message = f'{message} pre_course.'
 
-    course_cards, is_any_course_started, is_locked = get_course_card_information(test_user, [course])
+    course_cards, is_any_course_started, are_courses_completed = get_course_card_information(test_user, [course])
 
     assert course_cards == [{
         'course': course,
         'status': status,
         'grade': '',
-        'message': message
+        'message': message,
+        'pre_req': 'pre_course'
     }]
     assert not is_any_course_started
-    assert is_locked == is_course_locked
+    assert not are_courses_completed
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    'is_open_course, is_locked_course, is_unlocked_course',
+    [
+        (True, False, False),
+        (False, True, False),
+        (False, False, True),
+        (True, True, False),
+        (True, False, True),
+        (False, True, True),
+        (True, True, True),
+    ]
+)
+def test_get_omnipreneurship_courses_instructions(is_open_course, is_locked_course, is_unlocked_course):
+    """
+    Tests whether correct instructions are made for the courses depending on their statuses and whether they have
+    prerequisites courses or not
+    """
+    courses = []
+    open_courses = locked_courses = unlocked_courses = ''
+
+    if is_open_course:
+        courses.append({
+            'course': CourseOverviewFactory(display_name='open_course'),
+            'status': '',
+            'pre_req': ''
+        })
+        open_courses = 'You will have to complete <em>all</em> of the Omnipreneurship courses. However, you will ' \
+                       'have to enroll in and <em>complete open_course first.</em>'
+
+    if is_locked_course:
+        courses.append({
+            'course': CourseOverviewFactory(display_name='locked_course'),
+            'status': LOCKED,
+            'pre_req': 'pre_req_course'
+        })
+        locked_courses = 'Once you have completed pre_req_course and received a passing grade, the locked_course ' \
+                         'course will be unlocked. '
+
+    if is_unlocked_course:
+        courses.append({
+            'course': CourseOverviewFactory(display_name='un_locked_course'),
+            'status': NOT_STARTED,
+            'pre_req': 'pre_req_course'
+        })
+        unlocked_courses = 'un_locked_course has been unlocked for you to enroll in and complete! '
+
+    instructions = get_omnipreneurship_courses_instructions(courses)
+
+    assert instructions == {
+        'open_courses': open_courses,
+        'locked_courses': locked_courses,
+        'unlocked_courses': unlocked_courses
+    }
```

Comment:
It would be more readable and standard if we go separate these into expected_instructions and actual_instructions:

```
expected_instructions = {
        'open_courses': open_courses,
        'locked_courses': locked_courses,
        'unlocked_courses': unlocked_courses
}

actual_instructions = get_omnipreneurship_courses_instructions(courses)

assert expected_instructions == actual_instructions
```